Content.Shared/_Goobstation/Clothing/Systems/SharedSealableClothingSystem.cs

```diff
@@ -0,0 +1,446 @@
+        {
+            _popupSystem.PopupClient(Loc.GetString(comp.CurrentlySealedToggleFailPopup), uid, args.User);
+            _audioSystem.PlayPredicted(comp.FailSound, uid, args.User);
+            args.Cancel();
+
+            return;
+        }
+
+        return;
+    }
+    #endregion
+
+    /// <summary>
+    ///     Tries to start sealing process
+    /// </summary>
+    /// <param name="control"></param>
+    /// <returns></returns>
+    public bool TryStartSealToggleProcess(Entity<SealableClothingControlComponent> control, EntityUid? user = null)
+    {
+        var (uid, comp) = control;
+
+        // Prevent sealing/unsealing if modsuit don't have wearer or already started process
+        if (comp.WearerEntity == null || comp.IsInProcess)
+            return false;
+
+        if (user == null)
+            user = comp.WearerEntity;
+
+        var ev = new ClothingSealAttemptEvent(user.Value);
+        RaiseLocalEvent(control, ev);
+
+        if (ev.Cancelled)
+            return false;
+
+        // All parts required to be toggled to perform sealing
+        if (_toggleableSystem.GetAttachedToggleStatus(uid) != ToggleableClothingAttachedStatus.AllToggled)
+        {
+            _popupSystem.PopupClient(Loc.GetString(comp.ToggleFailedPopup), uid, user);
+            _audioSystem.PlayPredicted(comp.FailSound, uid, user);
+            return false;
+        }
+
+        // Trying to get all clothing to seal
+        var sealeableList = _toggleableSystem.GetAttachedClothingsList(uid);
+        if (sealeableList == null)
+            return false;
+
+        foreach (var sealeable in sealeableList)
+        {
+            if (!HasComp<SealableClothingComponent>(sealeable))
+            {
+                _popupSystem.PopupEntity(Loc.GetString(comp.ToggleFailedPopup), uid);
+                _audioSystem.PlayPredicted(comp.FailSound, uid, user);
+
+                comp.ProcessQueue.Clear();
+                Dirty(control);
+
+                return false;
+            }
+
+            comp.ProcessQueue.Enqueue(EntityManager.GetNetEntity(sealeable));
+        }
+
+        comp.IsInProcess = true;
+        Dirty(control);
+
+        NextSealProcess(control);
+
+        return true;
+    }
+
+    /// <summary>
+    ///     Recursively seals/unseals all parts of sealable clothing
+    /// </summary>
+    /// <param name="control"></param>
+    private void NextSealProcess(Entity<SealableClothingControlComponent> control)
+    {
+        var (uid, comp) = control;
+
+        // Finish sealing process
+        if (comp.ProcessQueue.Count == 0)
+        {
+            comp.IsInProcess = false;
+            comp.IsCurrentlySealed = !comp.IsCurrentlySealed;
+
+            if (comp.IsCurrentlySealed)
+                _audioSystem.PlayEntity(comp.SealCompleteSound, comp.WearerEntity!.Value, uid);
+            else
+                _audioSystem.PlayEntity(comp.UnsealCompleteSound, comp.WearerEntity!.Value, uid);
```

Comment:
Again, the same thing?
```suggestion
            _audioSystem.PlayEntity(comp.UnsealCompleteSound, comp.WearerEntity!.Value, uid);
```